src/Visualization/Python/PlotTrajectories.py

```diff
@@ -0,0 +1,155 @@
+                if "ApparentHorizons/ControlSystemAhA_Centers.dat" in file:
+                    A_data.append(
+                        np.array(
+                            file[
+                                "ApparentHorizons/ControlSystemAhA_Centers.dat"
+                            ][:]
+                        )
+                    )
+                if "ApparentHorizons/ControlSystemAhB_Centers.dat" in file:
+                    B_data.append(
+                        np.array(
+                            file[
+                                "ApparentHorizons/ControlSystemAhB_Centers.dat"
+                            ][:]
+                        )
+                    )
+        except Exception as e:
+            click.echo(f"Error reading file '{filename}': {e}")
+            return None, None
+
+    if A_data and B_data:
+        A = np.concatenate(A_data)
+        B = np.concatenate(B_data)
+        min_length = min(len(A), len(B))
+        return A[:min_length], B[:min_length]
+    else:
+        return None, None
+
+
+def plot_trajectory(AhA, AhB, plotlabel="", output=None, force=False):
+    indexs = [4, 5]  # 4 -> x, 5 -> y
+
+    if output and os.path.exists(output) and not force:
+        click.echo(
+            f"Warning: Output file '{output}' already exists. "
+            "Use --force to overwrite."
+        )
+        return
+
+    plt.figure(figsize=(10, 5))
+    plt.subplot(1, 2, 1)
+    plt.plot(
+        [AhA[i][indexs[0]] for i in range(len(AhA))],
+        [AhA[i][indexs[1]] for i in range(len(AhA))],
+        label="AhA",
+        color="blue",
+    )
+    plt.plot(
+        [AhB[i][indexs[0]] for i in range(len(AhB))],
+        [AhB[i][indexs[1]] for i in range(len(AhB))],
+        label="AhB",
+        color="red",
+    )
+    plt.xlabel("x")
+    plt.ylabel("y")
+    plt.title("Trajectories" + " " + plotlabel)
+    plt.legend()
+
+    plt.subplot(1, 2, 2)
+    x_diff = [AhA[i][indexs[0]] - AhB[i][indexs[0]] for i in range(len(AhA))]
+    y_diff = [AhA[i][indexs[1]] - AhB[i][indexs[1]] for i in range(len(AhA))]
+    plt.plot(x_diff, y_diff, color="black")
+    plt.xlabel("x1-x2")
+    plt.ylabel("y1-y2")
+    plt.title("Coordinate separation" + " " + plotlabel)
+
+    plt.tight_layout()
+
+    if output:
+        _, extension = os.path.splitext(output)
+        if extension.lower() not in [".pdf", ".png"]:
+            click.echo(
+                "Error: Output format not supported. "
+                "Please choose '.pdf' or '.png'."
+            )
+            return
+
+        if os.path.exists(output) and not force:
+            click.echo(
+                f"Error: Output file '{output}' already exists. "
+                "Use --force to overwrite."
+            )
+            return
+        plt.savefig(output)
+        click.echo(f"Plot saved as {output}")
+        plt.close()
+    else:
+        plt.show()  # No output file specified, display in graphic window
```

Comment:
This is taken care of by `@show_or_save_plot_command`, so you can delete this part :)